kani-compiler/src/codegen_cprover_gotoc/codegen/function.rs

```diff
@@ -245,33 +249,37 @@ impl<'tcx> GotocCtx<'tcx> {
                 self.tcx
                     .sess
                     .span_err(span, "The proof attribute cannot be applied to generic functions.");
+            } else {
+                let instance = Instance::mono(self.tcx, def_id);
+                if !self.fn_abi_of_instance(instance, ty::List::empty()).args.is_empty() {
+                    self.tcx
+                        .sess
+                        .span_err(span, "Functions used as harnesses can not have any arguments.");
+                }
             }
+        }
+    }
+
+    pub fn is_proof_harness(&self, def_id: DefId) -> bool {
```

Comment:
Since this function is called from e.g. reachability, it might not be appropriate to the put the check here. It may get called multiple times?

Reply:
It already was being checked twice. But I'm ok removing from here.